src/main/java/seedu/fridgefriend/food/Fridge.java

```diff
@@ -49,12 +52,15 @@ public void removeByIndex(int index) {
     }
 
     //@@author Vinci-Hu
-    private void editFoodQuantity(Food newFood, Food existingFood) {
-        int deltaQuantity = newFood.getQuantity();
-        int oriQuantity = existingFood.getQuantity();
-        int newQuantity = deltaQuantity + oriQuantity;
-        newFood.setQuantity(newQuantity);
-        existingFood.setQuantity(newQuantity);
+    private void editFoodQuantity(Food newFood, Food existingFood) throws InvalidQuantityException {
+        long deltaQuantity = newFood.getQuantity();
+        long oriQuantity = existingFood.getQuantity();
+        long newQuantity = deltaQuantity + oriQuantity;
+        if (newQuantity >= Integer.MAX_VALUE) {
```

Comment:
We might want to set the "MAX Value" not to integer max value, but rather a smaller value (perhaps 10k? 100k?) which would definitely prevent the case of Integer Overflows.